### PR TITLE
Feature/user profile edit

### PR DESCRIPTION
**Pull request by basantrijal**
profile update api

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -4,7 +4,8 @@
 
   namespace 'api' do
     namespace 'v1', :constraints => {format: 'json'} do
-      resources :registrations, :controller => "users", :only=> [:create]
+      resources :registrations, :controller => "users", :only => [:create]
```
Comment (allanxavierm) [inline config/routes.rb]: @basantrijal Why we need a registrations resource

File: app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -3,21 +3,29 @@ module V1
...
+        if user.update_attributes(user_params)
+          render json: { message: 'Your profile has been updated' }, status: :ok
+        else
+          render json: { error: user.errors }, status: :not_acceptable
+        end
+      end
 
-    end
+      def user_params
+        params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation, :avatar)
```
Comment (allanxavierm) [inline app/controllers/api/v1/users_controller.rb]: I don't think we need to populate password_confirmation to the database

File: app/controllers/api/v1/users_controller.rb [hunk 3]
```diff
@@ -3,21 +3,29 @@ module V1
...
       end
-    end
 
-    def user_params
-      params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation, :avatar)
-    end
+      def update
+        user = User.find(params[:id])
+        if user.update_attributes(user_params)
+          render json: { message: 'Your profile has been updated' }, status: :ok
```
Comment (allanxavierm) [inline app/controllers/api/v1/users_controller.rb:19]: These messages should go into en.yml

File: config/routes.rb (hunk 1 above)
Comment (basantrijal) [inline config/routes.rb]: I wanted to create a different route for user registration like api/v1/registrations independent of users.

File: app/controllers/api/v1/users_controller.rb (hunk 3 above)
Comment (basantrijal) [inline app/controllers/api/v1/users_controller.rb:19]: let's create a locale file after user controller is finalized.

File: config/routes.rb (hunk 1 above)
Comment (allanxavierm) [inline config/routes.rb]: why we need such a route

File: app/controllers/api/v1/users_controller.rb (hunk 3 above)
Comment (allanxavierm) [inline app/controllers/api/v1/users_controller.rb:19]: Don't wait to get anything finalized, these don't require significant effort to move into a locale file

Comment (basantrijal) [inline app/controllers/api/v1/users_controller.rb:19]: okay, i will create locale file in user-check-api branch which is latest branch

**Pull request closed without merging**